### PR TITLE
Don't use protocjar in mill.

### PR DESCRIPTION
**Pull request by sequencer**
https://github.com/os72/protoc-jar is a third-party repackage of protoc, last update was 13 months ago.
Latest jar was 3.11.4, however current version is 3.15.6 now.
This also fix a bug mentioned in #2160.
This PR fix this, and download binary based on the operation system directly.
Notice this PR is souce incompatible, should not be backported.

### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you update the FIRRTL spec to include every new feature/behavior?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you state the API impact?
- [ ] Did you specify the code generation impact?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- bug fix

#### API Impact

None

#### Backend Code Generation Impact

None

#### Desired Merge Strategy

- Rebase: You will rebase the PR onto master and it will be merged with a merge commit.

#### Release Notes

Compile with protoc directly in mill.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (1.2.x, 1.3.0, 1.4.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -119,13 +119,57 @@ class firrtlCrossModule(val crossScalaVersion: String) extends CrossSbtModule wi
...
+        else if (ppcle_64) "linux-ppcle_64"
+        else if (s390x) "linux-s390x"
+        else if (x86_32) "linux-x86_32"
+        else if (x86_64) "linux-x86_64"
+        else throw new Exception("mill cannot detect your architecture of your Linux")
+      else if (isWindows)
+        if (x86_32) "win32"
+        else if (x86_64) "win64"
+        else throw new Exception("mill cannot detect your architecture of your Windows")
+      else throw new Exception("mill cannot detect your operation system.")
```
Comment (sequencer) [inline build.sc:155]: Support all architecture in protobuf release: https://github.com/protocolbuffers/protobuf/releases

Comment (jackkoenig) [reply build.sc:155]: I think this is nicer than using that rando jar to download protoc. I do wish we could share this code between SBT and Mill but oh well 🤷

**Review by jackkoenig — APPROVED** [review]

**Comment by jackkoenig** [comment]
Need to apply this to SBT as well which is still using protoc 3.5.1

File: build.sc (hunk 1 above)
Comment (jackkoenig) [reply build.sc:155]: Not ARM 64-bit, MacOS ARM 64-bit which still supports x86_64 binaries via Rosetta 2. Most projects don't create native binaries for M1 yet.

Comment (jackkoenig) [reply build.sc:155]: That being said, this isn't supporting x64_64 Macs lol so need to fix that.

**Pull request merged**